internal/cloudapi/v2/v2.go

```diff
@@ -182,13 +187,31 @@ func (h *apiHandlers) PostCompose(ctx echo.Context) error {
 		}
 	}
 
+	packageSetKeys := imageType.PackageSetKeys()
+	packageSetsRepositories := make(map[string][]rpmmd.RepoConfig, len(packageSetKeys))
+
+	for _, packageSetKey := range packageSetKeys {
+		packageSetsRepositories[packageSetKey] = make([]rpmmd.RepoConfig, len(rcRepositories))
+		for j, repo := range rcRepositories {
+			packageSetsRepositories[packageSetKey][j].CheckGPG = repo.CheckGpg
+			packageSetsRepositories[packageSetKey][j].IgnoreSSL = repo.IgnoreSsl
+
+			if repo.Baseurl != nil {
+				packageSetsRepositories[packageSetKey][j].BaseURL = *repo.Baseurl
+			} else {
+				return echo.NewHTTPError(http.StatusBadRequest, "BaseUrl Must be specify")
```

Comment:
Please create a new error for this after https://github.com/osbuild/osbuild-composer/blob/main/internal/cloudapi/v2/errors.go#L38 and after https://github.com/osbuild/osbuild-composer/blob/main/internal/cloudapi/v2/errors.go#L97 as a 400, `ErrorNoBaseURLInPayloadRepository` or something. And then return it as `return HTTPError(ErrorNoBaseURLInPayloadRepository)`. 

For the message "BaseURL must be specified for payload repositories".